src/Parsing/Impl/Ast/ScopeDelegate.cs

```diff
@@ -0,0 +1,259 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Python.Core;
+using Microsoft.Python.Core.Collections;
+using Microsoft.Python.Core.Diagnostics;
+
+namespace Microsoft.Python.Parsing.Ast {
+    internal abstract class ScopeDelegate {
```

Comment:
I still think that `Delegate` is the wrong name to use for these. Delegates are a thing in C# already. I think `Info` was the better suffix.

Reply:
Ok I'll rename it